### ISSUE TITLE
BasicContext.loadLocale doesn't handle Windows locale names.

### ISSUE THREAD

**Issue by brcolow** [impossibl/pgjdbc-ng#285]
I have PostgreSQL 9.6 installed on my Windows 10 machine. The default locales listed in `postgres.conf` are as follows:

```
lc_messages = 'English_United States.1252'          # locale for system error message
lc_monetary = 'English_United States.1252'          # locale for monetary formatting
lc_numeric = 'English_United States.1252'           # locale for number formatting
lc_time = 'English_United States.1252'              # locale for time formatting
```

Furthermore it is not possible to change them to `en_US` on Windows, as this results in errors when starting Postgres:

```
2016-10-08 23:38:09 MST LOG:  invalid value for parameter "lc_monetary": "en_US.1252"
2016-10-08 23:38:09 MST LOG:  invalid value for parameter "lc_numeric": "en_US.1252"
2016-10-08 23:38:09 MST LOG:  invalid value for parameter "lc_time": "en_US.1252"
2016-10-08 23:38:09 MST FATAL:  configuration file "C:/Program Files/PostgreSQL/9.6/data/postgresql.conf" contains errors
```

Indeed the [Locale docs](https://www.postgresql.org/docs/9.6/static/locale.html) state (emphasis mine):

> What locales are available on your system under what names depends on what was provided by the operating system vendor and what was installed. On most Unix systems, the command locale -a will provide a list of available locales. **Windows uses more verbose locale names, such as German_Germany or Swedish_Sweden.1252, but the principles are the same.**

When attempting to connect to this postgres instance running with the default locales (English_United States.1252) the following error occurs:

```
Caused by: java.util.IllformedLocaleException: Ill-formed region: United States [at index 0]
        at java.util.Locale$Builder.setRegion(Locale.java:2495)
        at com.impossibl.postgres.system.BasicContext.loadLocale(BasicContext.java:285)
        at com.impossibl.postgres.system.BasicContext.init(BasicContext.java:265)
        at com.impossibl.postgres.jdbc.PGConnectionImpl.init(PGConnectionImpl.java:249)
        at com.impossibl.postgres.jdbc.ConnectionUtil.createConnection(ConnectionUtil.java:182)
        at com.impossibl.postgres.jdbc.AbstractDataSource.createConnection(AbstractDataSource.java:747)
        at com.impossibl.postgres.jdbc.PGDataSource.getConnection(PGDataSource.java:66)
        at com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:341)
        at com.zaxxer.hikari.pool.HikariPool.checkFailFast(HikariPool.java:506)
        ... 4 more
```

The reason is in fact quite apparent - it is these lines in [BasicContext.loadLocale()](https://github.com/impossibl/pgjdbc-ng/blob/develop/src/main/java/com/impossibl/postgres/system/BasicContext.java#L281):

``` java
String[] localeIds = localeSpec.split("_|\\.");

switch (row.getColumn(0).toString()) {
  case "lc_numeric":
    Locale numLocale = new Locale.Builder().setLanguageTag(localeIds[0]).setRegion(localeIds[1]).build();
    decimalFormatter = (DecimalFormat) DecimalFormat.getNumberInstance(numLocale);
    decimalFormatter.setParseBigDecimal(true);
    currencyFormatter = (DecimalFormat) NumberFormat.getCurrencyInstance(numLocale);
    currencyFormatter.setParseBigDecimal(true);
    break;
  case "lc_time":
    Locale timeLocale = new Locale.Builder().setLanguageTag(localeIds[0]).setRegion(localeIds[1]).build();
}
```

The input String "English_United States.1252" is split up thusly:

["English", "United States", "1252"]

which is then fed to `Locale.Builder` like so:

``` java
Locale numLocale = new Locale.Builder().setLanguageTag("English").setRegion("United States").build();
```

However, "United States" is not a valid region specifier, as Java's [Locale docs](https://docs.oracle.com/javase/8/docs/api/java/util/Locale.html#def_region) state:

> ISO 3166 alpha-2 country code or UN M.49 numeric-3 area code. You can find a full list of valid country and region codes in the IANA Language Subtag Registry (search for "Type: region"). The country (region) field is case insensitive, but Locale always canonicalizes to upper case.
> 
> Well-formed country/region values have the form [a-zA-Z]{2} | [0-9]{3}
> 
> Example: "US" (United States), "FR" (France), "029" (Caribbean)

To fix this issue we essentially need to create a mapping between the Window's locales (those understood by Postgres on Windows) and the Unix locales (those understood by Java).

A full (?) list of the Window's locales can be found [here](https://msdn.microsoft.com/en-us/goglobal/bb895996.aspx).

I would be more than happy to create a PR along those lines (essentially just create an immutable hash map mapping the Windows locale String's to the unix locale strings) if you are okay with this approach.

Let me know what you guys think. Thanks.

**Comment by kdubb** [comment]
Good catch! Grateful for the research & explanation.  Your solution sounds acceptable and would love a PR.

**Comment by brcolow** [comment]
@kdubb PR opened: https://github.com/impossibl/pgjdbc-ng/pull/287

**Comment by jesperpedersen** [comment]
Fixed by #287
